installer/installer.py

```diff
@@ -522,6 +537,98 @@ def cmd_install_netmodules(public_slave=False):
     return
 
 
+
+def cmd_install_cni(public_slave,
+                    cni_plugins_dir,
+                    cni_conf_dir):
+    """
+    Install Calico's CNI plugin.  A successful completion of the task
+    indicates successful installation
+
+    :param public_slave: Flag indicating if this is a public slave.
+    """
+    # Load the current Calico install info for CNI
+    install_config = load_config(CNI_INSTALL_CONFIG)
+
+    # Before starting the install, check that we are able to locate the agent
+    # process - if not there is not much we can do here.
+    if not install_config:
+        _log.debug("Have not started installation yet")
+        # noinspection PyTypeChecker
+        agent_process = wait_for_process(AGENT_EXE_RE,
+                                         AGENT_EXE_PARMS_RE,
+                                         MAX_TIME_FOR_AGENT_RESTART,
+                                         fail_if_not_found=False)
+        if not agent_process:
```

Comment:
I suppose the user would be confused as to why it succeeded. I'll make it sys.exit(1) so it repeatedly fails, leaving it to the operator to diagnose.